pkg/resources/procedure.go

```diff
@@ -548,11 +530,18 @@ func ReadContextProcedure(ctx context.Context, d *schema.ResourceData, meta inte
 	for i, arg := range args {
 		argTypes[i] = arg.(map[string]interface{})["type"].(string)
 	}
-	procedureDetails, err := client.Procedures.Describe(ctx, sdk.NewDescribeProcedureRequest(id.name, id.argTypes))
+	procedureDetails, err := client.Procedures.Describe(ctx, sdk.NewDescribeProcedureRequest(id.WithoutArguments(), id.Arguments()))
 	if err != nil {
 		// if procedure is not found then mark resource to be removed from state file during apply or refresh
 		d.SetId("")
-		return nil
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary:  "Describe procedure failed.",
+				// TODO: link to the design decisions doc
+				Detail: "See our document on design decisions for procedures: <LINK (coming soon)>",
```

Comment:
Please change warning descriptions and remove TODO, because right now I don't think we have a design doc coming for procedures (was it a copy-paste error from grant resources ?). IMO just `d.Id()` would be sufficient to spot the error.

Reply:
fixed